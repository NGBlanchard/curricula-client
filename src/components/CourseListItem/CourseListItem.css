.CourseListItem {
  background: none;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  border: 1px solid #3f455038;
  margin-bottom: 10px;
  font-size: smaller;
  display: block;
  padding: 10px 10px 0;
  text-align: left;
  width: 100%;
  text-decoration: none;
  height: 150px;
}

.CourseListItem:hover {
  cursor: pointer;
  background-color: #f7f8f9;
  box-shadow: 0 4px 8px 0 rgba(202, 80, 255, 0.19);
}

.CourseListItem,
.CourseListItem:visited {
  color: #555;
}

.CourseListItem__header {
  align-items: baseline;
  display: flex;
  justify-content: space-between;
}
.CourseListItem__heading {
  margin: 0;
}

.CourseListItem__author {
  font-weight: bold;
  color: #CA1551;
}

.CourseListItem__date {
  display: none;
}

.CourseListItem__footer {
  padding: 10px 0;
}

.CourseListItem__comment-count {
  float: right;
  font-size: 1.3em;
}

.CourseListItem__comment-count .fa-layers-text {
  left: 50%;
  top: 44%;
  font-family: Arial;
  font-weight: bold;
  font-size: smaller;
}

@media screen and (min-width: 440px) {
  .CourseListItem__date {
    display: initial;
  }
}
