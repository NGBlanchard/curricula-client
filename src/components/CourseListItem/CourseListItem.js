import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { NiceDate } from '../Utils/Utils'
import CourseApiService from '../../services/course-api-service'
import CourseContext from '../../context/CourseContext'

import './CourseListItem.css'


export default class CourseListItem extends Component {
  state = {
    user: "",
  }
  static contextType = CourseContext

    componentDidMount = () => {
      // console.log(this.props.course.author)
      this.context.clearError()
      CourseApiService.getUserById(this.props.course.author)
      .then(res =>
        this.setState({
          user: res.user_name
        })
      )}

  render() {
    const { course } = this.props
    return (
      <Link to={`/course/${course.id}`} className='CourseListItem'>
        <header className='CourseListItem__header'>
          <h2 className='CourseListItem__heading'>
            {course.title}
          </h2>
          <CourseDate course={course} />
        </header>
        <footer className='CourseListItem__footer'>
          <CourseStyle course={course} />
          <span className='CourseListItem__author'>
            <p className='CourseListItem__author'>
              Created by  {this.state.user}
            </p>
          </span>
        </footer>
      </Link>
    )
  }
}

function CourseStyle({ course }) {
  return (
    <span className='CourseListItem__style'>
      Topic : {course.topic}
    </span>
  )
}

function CourseDate({ course }) {
  return (
    <span className='CourseListItem__date'>
      <NiceDate
        date={course.date_created}
      />
    </span>
  )
}