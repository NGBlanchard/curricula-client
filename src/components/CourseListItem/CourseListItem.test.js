import CourseListItem from './CourseListItem'
import { BrowserRouter as Router } from "react-router-dom";
import React from 'react'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'


configure({ adapter: new Adapter() });

describe('<CourseListItem />', () => {
  const course = {
    id: 1,
    title: 'title',
    description: 'description',
    readings: 'readings',
    notes: 'notes',
    topic: 'topic',
    duration: 12,
    date_created: "2019-09-28T16:43:22.991Z",
    author_id: 2
  }

const context = {
  clearError(){
    this.setState({ error: null })
  } 
}
  it('Renders without crashing', () => {
    mount(
      <Router>
        <CourseListItem
          key={course.id}
          course={course}
        />
      </Router>, context)
    })
  it('renders the item by default', () => {
    const wrapper = mount(
      <Router>
        <CourseListItem 
          key={course.id} 
          course={course}
        />
      </Router>, context)
    expect(toJson(wrapper)).toMatchSnapshot()  
  });
})
