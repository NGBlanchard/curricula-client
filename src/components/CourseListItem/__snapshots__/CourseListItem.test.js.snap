// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseListItem /> renders the item by default 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CourseListItem
      course={
        Object {
          "author_id": 2,
          "date_created": "2019-09-28T16:43:22.991Z",
          "description": "description",
          "duration": 12,
          "id": 1,
          "notes": "notes",
          "readings": "readings",
          "title": "title",
          "topic": "topic",
        }
      }
      key="1"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="CourseListItem"
        to="/course/1"
      >
        <Route
          path="\\\\/course\\\\/1"
        >
          <Link
            aria-current={null}
            className="CourseListItem"
            replace={false}
            to="/course/1"
          >
            <a
              aria-current={null}
              className="CourseListItem"
              href="/course/1"
              onClick={[Function]}
            >
              <header
                className="CourseListItem__header"
              >
                <h2
                  className="CourseListItem__heading"
                >
                  title
                </h2>
                <CourseDate
                  course={
                    Object {
                      "author_id": 2,
                      "date_created": "2019-09-28T16:43:22.991Z",
                      "description": "description",
                      "duration": 12,
                      "id": 1,
                      "notes": "notes",
                      "readings": "readings",
                      "title": "title",
                      "topic": "topic",
                    }
                  }
                >
                  <span
                    className="CourseListItem__date"
                  >
                    <NiceDate
                      date="2019-09-28T16:43:22.991Z"
                    >
                      28th September 2019
                    </NiceDate>
                  </span>
                </CourseDate>
              </header>
              <footer
                className="CourseListItem__footer"
              >
                <CourseStyle
                  course={
                    Object {
                      "author_id": 2,
                      "date_created": "2019-09-28T16:43:22.991Z",
                      "description": "description",
                      "duration": 12,
                      "id": 1,
                      "notes": "notes",
                      "readings": "readings",
                      "title": "title",
                      "topic": "topic",
                    }
                  }
                >
                  <span
                    className="CourseListItem__style"
                  >
                    Topic : 
                    topic
                  </span>
                </CourseStyle>
                <span
                  className="CourseListItem__author"
                >
                  <p
                    className="CourseListItem__author"
                  >
                    Created by  
                  </p>
                </span>
              </footer>
            </a>
          </Link>
        </Route>
      </NavLink>
    </CourseListItem>
  </Router>
</BrowserRouter>
`;
